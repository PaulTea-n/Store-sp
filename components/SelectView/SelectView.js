import React, { useState, useContext } from "react";
import SelectViewContext from "../../contexts/SelectViewContext/SelectViewContext";
import styles from "./SelectView.module.scss";

const SelectView = () => {
  const { currentView, setCurrentView } = useContext(SelectViewContext);

  return (
    <div className={styles.select}>
      <select
        name="view"
        id="view"
        value={currentView}
        onChange={({ target: { value } }) => setCurrentView(value)}
      >
        <option value="TABLE">TABLE</option>
        <option value="CARDS">CARDS</option>
      </select>
    </div>
  );
};

export default SelectView;
