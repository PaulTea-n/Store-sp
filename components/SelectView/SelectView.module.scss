.select {
    background-color: #7A7A7A;
    height: 40px;
    padding-top: 20px;
}