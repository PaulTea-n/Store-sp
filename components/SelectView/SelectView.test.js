import SelectView from "./SelectView";
import { render } from "@testing-library/react";

describe("SelectView component working", () => {
  test("should render working", () => {
    render(
      <SelectView name="view" id="view" value="test" onChange={() => {}} />
    );
  });

  test("should SelectView fragment match snapshot", () => {
    const { asFragment } = render(
      <SelectView name="view" id="view" value="test" onChange={() => {}} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
