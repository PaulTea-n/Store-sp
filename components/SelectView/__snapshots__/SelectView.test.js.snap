// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectView component working should SelectView fragment match snapshot 1`] = `
<DocumentFragment>
  <div
    class="select"
  >
    <select
      id="view"
      name="view"
    >
      <option
        value="TABLE"
      >
        TABLE
      </option>
      <option
        value="CARDS"
      >
        CARDS
      </option>
    </select>
  </div>
</DocumentFragment>
`;
